ansible_events/drools/durable_rules_engine.py

```diff
@@ -0,0 +1,235 @@
+import json
+import logging
+import os
+
+import requests
+
+_logger = logging.getLogger(__name__)
+
+
+class DurableRulesEngine:
+    def __init__(self, host):
+        self._host = host
+        self._last_response = None
+
+    def create_ruleset(self, ruleset_name, ruleset_string):
+        request = {ruleset_name: json.loads(ruleset_string)}
+
+        response = requests.post(
+            self._host + "/create-durable-rules-executor", json=request
+        )
+        if response.status_code != 200:
+            raise Exception(
+                "Invalid status code: "
+                f"{response.status_code} - {response.reason}\n"
+                + json.loads(response.content)["details"]
+            )
+
+        return response.text
+
+    def assert_event(self, session_id, serialized_fact):
+        _logger.warning("assert_event not yet implemented: using assert_fact")
+        return self.assert_fact(session_id, serialized_fact)
+
+    def assert_fact(self, session_id, serialized_fact):
+        fact = json.loads(serialized_fact)
+        if "j" in serialized_fact:
+            fact["j"] = 1
+
+        serialized_fact = json.dumps(fact)
+
+        response = requests.post(
+            f"{self._host}/rules-durable-executors/{session_id}/process",
+            json=json.loads(serialized_fact),
+        )
+
+        if response.status_code != 200:
+            raise Exception(
+                "Invalid status code: "
+                f"{response.status_code} - {response.reason}\n"
+                + json.loads(response.content)["details"]
+            )
+
+        self._last_response = response.json()
+        # drools returns the list in the opposite order
+        self._last_response.reverse()
+
+        return (0, session_id)
+
+    def start_action_for_state(self, handle):  # real signature unknown
+
+        if self._last_response:
+            resp = self._last_response.pop()
+        else:
+            return None
+
+        return ('{ "sid":"0", "id":"sid-0", "$s":1}', json.dumps(resp), handle)
+
+    def complete_and_start_action(self, handle):  # real signature unknown
+        _logger.info("complete_and_start_action: %d", handle)
```

Comment:
@evacchi The %d should be %s handle has a type of str
<class 'str'>

Reply:
fixed